nvdaHelper/vbufBackends/mshtml/mshtml.cpp

```diff
@@ -1007,6 +1010,13 @@ if(!(formatState&FORMATSTATE_INSERTED)&&nodeName.compare(L"INS")==0) {
 	if(nodeName.compare(L"OL")==0||nodeName.compare(L"UL")==0) {
 		//Ordered lists should number their list items
 		LIIndex=1;
+		// set the list index if list has start attribute
+		if ((tempIter = attribsMap.find(L"HTMLAttrib::start")) != attribsMap.end()) {

```

Comment:
While I think this is done a fair bit in this file. Please don't assign a value and make a comparison all in one. It's hard to read and error prone. Please split this into two lines, you can also then check for empty:
```
tempIter = attribsMap.find(L"HTMLAttrib::start")
if (tempIter != attribsMap.end() && !tempIter->second.empty()) {
```